dev-aws/kafka-shared-msk/energy-budget-plan/energy-budget-plan.tf

```diff
@@ -0,0 +1,71 @@
+resource "kafka_topic" "eqdb_loader" {
+  name               = "energy-budget-plan.eqdb-loader"
+  replication_factor = 3
+  partitions         = 1
+  config = {
+    # retain 100MB on each partition
+    "retention.bytes" = "104857600"
+    # keep data for 18 hours
+    "retention.ms" = "64800000"
+    # allow max 1 MB for a message
+    "max.message.bytes" = "1048576"
+    "compression.type"  = "zstd"
+    "cleanup.policy"    = "delete"
+  }
+}
+
+resource "kafka_topic" "budget_plan" {
+  name               = "energy-budget-plan.budget-plan"
+  replication_factor = 3
+  partitions         = 10
+  config = {
+    # Use tiered storage
+    "remote.storage.enable" = "true"
+    # keep data for 7 years
+    "retention.ms" = "220898482000"
+    # keep data in hot storage for 2 days
+    "local.retention.ms" = "172800000"
+    # allow max 1 MB for a message
+    "max.message.bytes" = "1048576"
+    "compression.type"  = "zstd"
+    "cleanup.policy"    = "delete"
+  }
+}
+
+resource "kafka_topic" "customer_change" {
+  name               = "energy-budget-plan.customer-change"
+  replication_factor = 3
+  partitions         = 10
+  config = {
+    # Use tiered storage
+    "remote.storage.enable" = "true"
+    # keep data for 2 years
+    "retention.ms" = "63113852000"
+    # keep data in hot storage for 7 days
+    "local.retention.ms" = "604800000"
+    # allow max 1 MB for a message
+    "max.message.bytes" = "1048576"
+    "compression.type"  = "zstd"
+    "cleanup.policy"    = "delete"
+  }
+}
+
+module "eqdb_loader_process" {
+  source           = "../../../modules/tls-app"
+  produce_topics   = [kafka_topic.eqdb_loader.name]
+  cert_common_name = "energy-budget-plan/eqdb-loader"
+}
+
+module "budget_plan_fabricator" {
+  source           = "../../../modules/tls-app"
+  consume_topics   = { (kafka_topic.eqdb_loader.name) : "energy-budget-plan.eqdb-fabricator-v1" }
+  produce_topics   = [kafka_topic.customer_change.name]
+  cert_common_name = "energy-budget-plan/eqdb-fabricator"
+}
```

Comment:
The way this is currently built it consumes its own messages. We can change that of course but maybe we add in consume on the `customer_change` too?

Reply:
Sure, can do that while I am here.

Reply:
@utilitywarehouse/pubsub do consumer group names have to be unique or can I specify the same consumer group name on different topics?

Reply:
Hmmm, this is a good question. The way the tls-app is module is working right now, doing that would create the group_acl resource twice https://github.com/utilitywarehouse/kafka-cluster-config/blob/main/modules/tls-app/main.tf#L13-L21, so I think Terraform would be confused bu that.

I'll look into it and see if we can adjust that, but for now I would suggest a different consumer group.

Reply:
I've made the consumer groups unique for now.